lib/plato.js

```diff
@@ -118,86 +131,89 @@ function inspect(files, outputDir, options, done) {
 
 	var commonBasePath = util.findCommonBase(files);
 
-	function runReports(files, done) {
-		files.forEach(function reportEachFile(file) {
-			if (options.exclude && options.exclude.test(file)) {
-				return;
-			}
-
-			if (options.recurse && fs.statSync(file).isDirectory()) {
-				files = fs.readdirSync(file).map(function resolveFile(innerFile) {
-					return path.join(file, innerFile);
-				});
-				runReports(files);
-			} else if (file.match(/\.(js|jsx|mjs|es6|ts|tsx)$/)) {
-				log.info('Reading "%s"', file);
-
-				var fileShort = file.replace(commonBasePath, '');
-				var fileSafe = fileShort.replace(/[^a-zA-Z0-9]/g, '_');
-				var source = fs.readFileSync(file).toString();
-				var trimmedSource = source.trim();
-				if (!trimmedSource) {
-					log.info('Not parsing empty file "%s"', file);
-					return;
-				}
-
-				// if skip empty line option
-				if (options.noempty) {
-					source = source.replace(/^\s*[\r\n]/gm, '');
-				}
-
-				// if begins with shebang
-				if (source[0] === '#' && source[1] === '!') {
-					source = '//' + source;
-				}
-				var report = {
-					info: {
-						file: file,
-						fileShort: fileShort,
-						fileSafe: fileSafe,
-						link: fileDir + '/' + fileSafe + '/index.html'
-					}
-				};
-
-				var error = false;
-				_.each(reporters, function processReporter(reporter, name) {
-					if (!flags[name]) {
-						return;
-					}
-					try {
-						report[name] = reporter.process(source, flags[name], report.info);
-					} catch (e) {
-						error = true;
-						log.error('Error reading file %s: ', file, e.toString());
-						log.error(e.stack);
-					}
-				});
-
-				if (error) {
-					return;
-				}
-				reports.push(report);
-				if (fileOutputDir) {
-					var outdir = path.join(fileOutputDir, report.info.fileSafe);
-					if (!fs.existsSync(outdir)) {
-						fs.mkdirSync(outdir);
-					}
-					writeFileReport(outdir, report, source, options);
-				}
-			}
-		});
-		if (done) {
-			done();
-		}
-	}
-
-	if (!fileOutputDir) {
-		runReports(files, function() {
+  async function runReports(files, done) {
```

Comment:
Minor suggestion, but breaking out some utility functions to break this long method up might be nice.

Reply:
@briandonahue, totally agree with you, but I think for now we are trying to come up with more required features, and then we can polish this and rebuild, it requires a lot.